source/dub/packagemanager.d

```diff
@@ -71,14 +72,30 @@ class PackageManager {
 		auto ret = appender!(NativePath[])();
 		ret.put(cast(NativePath[])m_searchPath); // work around Phobos 17251
 		if (!m_disableDefaultSearchPaths) {
-			ret.put(cast(NativePath[])m_repositories[LocalPackageType.user].searchPath);
-			ret.put(cast(NativePath)m_repositories[LocalPackageType.user].packagePath);
-			ret.put(cast(NativePath[])m_repositories[LocalPackageType.system].searchPath);
-			ret.put(cast(NativePath)m_repositories[LocalPackageType.system].packagePath);
+			foreach (ref repo; m_repositories) {
+				ret.put(cast(NativePath[])repo.searchPath);
+				ret.put(cast(NativePath)repo.packagePath);
+			}
 		}
 		return ret.data;
 	}
 
+	/** Sets additional (read-only) package cache paths to search for packages.
+
+		Cache paths have the same structure as the default cache paths, such as
+		".dub/packages/".
+	*/
+	@property void customCachePaths(NativePath[] custom_cache_paths)
+	{
+		import std.algorithm.iteration : map;
+		import std.array : array;
+
+		m_repositories.length = LocalPackageType.max+1;
```

Comment:
Might be worth pointing out that this removes all previously set custom cache paths (though it gets kinda implied by "set").